migration/migration_utils.py

```diff
@@ -32,114 +33,208 @@ def get_process_state(process, topology, initial_state):
+    if process_class.__name__ == 'Complexation':
+        from arrow import StochasticSystem
+        process.system = StochasticSystem(process.stoichiometry, random_seed=0)
+    # Metabolism requires gtp_to_hydrolyze
+    elif process_class.__name__ == 'Metabolism':
+        updates_file = f'data/migration/process_updates_t{init_time}.json'
+        with open(updates_file, 'r') as f:
+            proc_updates = json.load(f)
+        gtp_hydro = proc_updates['PolypeptideElongation']['process_state'][
+            'gtp_to_hydrolyze']
+        initial_state['process_state'] = {'polypeptide_elongation': {
+            'gtp_to_hydrolyze': gtp_hydro}}
+    # MassListener needs initial masses to calculate fold changes
+    elif process_class.__name__ == 'MassListener':
+        t0_file = 'data/migration/wcecoli_t0.json'
+        with open(t0_file, 'r') as f:
+            t0_data = json.load(f)
+        t0_mass = t0_data['listeners']['mass']
+        process.first_time_step = False
+        process.dryMassInitial = t0_mass['dry_mass']
+        process.proteinMassInitial = t0_mass['protein_mass']
+        process.rnaMassInitial = t0_mass['rna_mass']
+        process.smallMoleculeMassInitial = t0_mass['smallMolecule_mass']
+        process.timeInitial = 0
+
+    if partition:
+        # run the process and get an update
+        actual_request, actual_update = run_partitioned_process(
+            process, process_class.topology,
+            initial_time=init_time, initial_state=initial_state)
+
+        # Compare requested bulk counts
+        with open(f"data/migration/bulk_requested_t{init_time}.json", 'r') as f:
+            wc_request = json.load(f)
+        assert np.all(actual_request == wc_request[process_class.__name__])
+    else:
+        actual_update = run_non_partitioned_process(
+            process, process_class.topology,
+            initial_time=init_time, initial_state=initial_state)
+    # Compare unique molecule updates
+    with open(f"data/migration/process_updates_t{init_time}.json", 'r') as f:
+        wc_update = json.load(f)
+    if process_class.__name__ in wc_update:
+        # wcEcoli and viv-ecoli generate unique indices very differently
+        assert recursive_compare(actual_update, wc_update[process_class.__name__],
+            check_keys_strict=False, ignore_keys={'unique_index', 'RNAP_index'})
+    # Compare listener values
+    with open(f"data/migration/wcecoli_listeners_t{init_time}.json", 'r') as f:
+        wc_listeners = json.load(f)
+    actual_listeners = actual_update.get('listeners', {})
+    # Certain listeners are at higher level in viv-ecoli
+    if 'monomer_counts' in actual_listeners:
+        actual_listeners = {'monomer_counts': actual_listeners}
+    if 'mRNA_counts' in actual_listeners:
+        actual_listeners = {'mrna_counts': actual_listeners}
+    # Gene copy number determined by listener after everything updates in wcEcoli
+    # TODO: Move this out of TfBinding process
```

Comment:
Turns out the gene copy number listener that we've implemented in `TFBinding` lags one timestep behind that reported by the equivalent wcEcoli listener. I'll create a new Step to handle this.